### DISCUSSION TITLE
Recycle Bin Custom View and WindowManager

### DISCUSSION THREAD

**Discussion by PurSpyk** [LesFerch/WinSetView#31] [General]
Amazing program. One request, would it be possible to add Recycle Bin to the folders you can customise?

**Comment by LesFerch** [comment]
Special folders, such as **Recycle Bin**, cannot be configured via the **FolderTypes** registry key, so all I could add to the interface is a drop down to select it's default view. Since it's default view is already **Details**, I didn't see any point in providing an option to change that.

However, WinSetView supports capturing custom view settings for the Recycle Bin, Control Panel, etc. via a **WinSetViewCustom.reg** file which can be created using the **CaptureCustom.vbs** script. It's described in detail in the manual. Please check that out and let me know if you need any assistance with it.

I have tested changing the default sort order for the Recycle Bin from **Name** to **Date Deleted** via this method and it works great. What settings did you want to change?

**Reply by PurSpyk** [reply]
I want to set the column spacing globally for the Recycle Bin. I find if I change the columns spacings in File Explorer it always defaults beck to some "original" settings. Thats why I thought there was some registry settings being used each time. I have large monitors so the default spacings are not ideal.

**Comment by LesFerch** [comment]
By "column spacing" do you mean "column widths"? If so, yes, you can capture your preferred column widths using the **CaptureCustom.vbs** script.

When you run WinSetView, it clears the Bags/BagMRU registry keys, which is why you are seeing Recycle Bin return to default settings. By adding your preferred settings (that you capture using CaptureCustom.vbs) you will retain your preferred Recycle Bin settings.

**Reply by PurSpyk** [reply]
Ok, will give it a go, but will the widths be remembered?

**Reply by LesFerch** [reply]
The general answer is yes, but because they are the somewhat volatile Bags/BagMRU keys, a Windows update _may_ revert them to defaults. If that happens, you can run WinSetView again to reapply the captured settings.

**Comment by PurSpyk** [comment]
Went through the steps, created the reg file, moved the file but my Recycle bin always defaults to some odd setting.  If I customise some of the app views in Control Panel, ie Program and Features, I set this to medium icons, is this supposed to be saved in the reg file the capturecustom.vbs created?

**Comment by LesFerch** [comment]
Yes, capturecustom.vbs does capture the Control Panel views,

Please note, _without_ a WinSetViewCustom.reg file, Recycle Bin will return to default settings whenever WinSetView is run, but Control Panel will retain it's settings. _With_ a WinSetViewCustom.reg file, Recycle Bin and Control Panel will have the settings that you captured when you ran capturecustom.vbs.

Capturecustom.vbs captures _all_ Bags/BagMRU settings, so it should be run _immediately_ after running WinSetView (which clears those keys) and setting the extra views (such as Recycle Bin) that you want to add. That way you get a minimum number of settings captured and the most predictable results.

**Comment by LesFerch** [comment]
It sounds like the WinSetViewCustom.reg file is not being applied. Possible reasons: file named incorrectly, file not in the active appdata directory. Where are you running WinSetView from? If it's in the Program Files, Program Files (x86) folder, or read-only network drive, appdata will not be the directory you might think. Try clicking Load Settings (Advanced interface) to see where the settings are located.

**Comment by LesFerch** [comment]
Your WinSetViewCustom.reg file is too big. It should only be about 7K. You have to do it like this:

1. Delete that WinSetViewCustom.reg file. It's not good.
2. Run WinSetView and click Submit to set your preferred defaults.
3. Do NOT open any other folders!
4. Open Recycle Bin and set the options you want
5. Run CaptureCustom.vbs
6. Verify that CaptureCustom.reg is smallish (mine is 7K).
7. Rename CaptureCustom.reg to WinSetViewCustom.reg and place it in the WinSetVew AppData folder
8. Run WinSetView again, click Submit, and then verify that Recycle Bin has the correct settings

**Comment by PurSpyk** [comment]
Followed your instructions, but the reg file is now 306KB. No Windows open, and I even rebooted before opening Recycle Bin and running the script

**Comment by LesFerch** [comment]
OK, I'm perplexed. What would cause your Bags/BagMRU keys to get immediately populated with so many entries? If you haven't already done so, please try it again _without_ rebooting.

If the file is still big without rebooting, then I think something is set on your machine to run when Explorer starts that causes a whole lot of folders to be opened (or at least behave like they've been opened). The Bags/BagMRU key should be very close to empty after running WinSetView and clicking Submit. Those keys normally get populated bit by bit as you open folders in File Explorer.

**Comment by PurSpyk** [comment]
I have many windows customised in size etc, I use a program called WindowManager that saves window size and positions. I like all my apps opening at the same place, and each app. window has a desired size. I am OCD must be because I am a Civil Engineer, like things to be ordered.

**Comment by LesFerch** [comment]
Ahh, so culprit found! You need to temporarily disable WindowManager so you have a clean registry to capture.
